### DISCUSSION TITLE
which account gets the Promo code ? -Solved

### DISCUSSION THREAD

**Discussion by omidRMR** [masterking32/MCF_HamsterKombat#8] [Q&A] [answered]
Hi, 
Apparently, the process of getting a promo key is executed independent of buying upgrades. 

![image](https://github.com/user-attachments/assets/f7f47468-909a-4b2a-bcdf-40939e07b5b0)

As you can see, checking accounts is completed and the bot has found a new key during the rest time. Now which account would get this key?

**Comment by Fy0urM** [comment]
Keys are bound to the ip on which they were received, and in the next cycle the first account with the same ip that was when the key was received uses this key

**Reply by omidRMR** [reply]
Thank you. 
I have 5 accounts and checking and upgrading them ususally takes less than 5 minutes. So the key is usually generated in the rest time. Since I don't use a proxy, the first account (first session) in the next round would get the key, all the time.  Am I right?

**Reply by Fy0urM** [reply, marked as answer]
Yes, if all sessions have 1 ip, they will receive keys one by one, whichever session will be the first to receive a key when the validation cycle starts. But it can't get more keys than it should. I hope so)
